push-notifications/android/src/main/java/com/capacitorjs/plugins/pushnotifications/NotificationChannelManager.java

```diff
@@ -140,35 +137,4 @@ public void listChannels(PluginCall call) {
             call.unavailable();
         }
     }
-
-    /**
-     * Create notification channel
-     */
-    public void createForegroundNotificationChannel() {
```

Comment:
@jcesarmobile Do we still need this, or [is adding a default channel id ](https://firebase.google.com/docs/cloud-messaging/android/client#manifest)to the manifest enough?

Reply:
As far as I am aware, it's not needed anymore. The `CommonNotificationBuilder.getOrCreateChannel(` will take care of upserting the channel.

The Firebase SDK will execute similar logic as what's explain in #1649:

It will derive the `channelId` to be used for the foreground in the following order:

1. Firstly it will check if the incoming notification has a `channelId` set
2. Then it will check for a possible given value in the `AndroidManifest.xml`<sup>1</sup>
3. Lastly it will use the fallback `channelId` that the Firebase SDK provides for us. This channel will be created by the Firebase SDK upon receiving the first push message<sup>2</sup>

<sup>1</sup> From the [Firebase docs](https://firebase.google.com/docs/cloud-messaging/android/client#:~:text=(Optional)%20From%20Android,Manage%20notification%20channels):

> From Android 8.0 (API level 26) and higher, notification channels are supported and recommended. FCM provides a default notification channel with basic settings. If you prefer to create and use your own default channel, set `default_notification_channel_id` to the ID of your notification channel object as shown; FCM will use this value whenever incoming messages do not explicitly set a notification channel.

(see [README](https://github.com/firebase/quickstart-android/tree/master/messaging#set-default-channel) also)

<sup>2</sup> This is similar to how `react-native-push-notification` is handling this: https://github.com/zo0r/react-native-push-notification/blob/master/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationConfig.java#LL69C19-L69C50

I think this makes way more sense than the current solution. It will also automatically inherit the channel settings set for the incoming notification. So that's more expected behaviour for the developer

---

It would make sense to explain all this to the user of course. I did add some stuff to the docs in my other PR (https://github.com/ionic-team/capacitor-plugins/pull/1649/files#diff-09c58e1cc87af661e7a9d0081a9360efcd3da00a53911c3f6b998a66168ea170). Honestly, I don't know why I didn't add that to this PR. Probably just forgot about it. I can add it if you want?

Reply:
Yes, I would add that note about the default notification channel in case users needed to define and use their own.  Thank yoU!

Reply:
Added! I tried my best to explain it as clear as possible.